src/main/java/com/couchbase/lite/auth/TokenStoreFactory.java

```diff
@@ -0,0 +1,28 @@
+//
+// Copyright (c) 2016 Couchbase, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+// except in compliance with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the
+// License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+// either express or implied. See the License for the specific language governing permissions
+// and limitations under the License.
+//
+package com.couchbase.lite.auth;
+
+import android.content.Context;
+import android.os.Build;
+
+/**
+ * Created by hideki on 6/22/16.
+ */
+public class TokenStoreFactory {
```

Comment:
To create OpenIDConnectAuthenticator, it requires TokenStore in addition to callback.
```
public static Authenticator createOpenIDConnectAuithenticator(OpenIDConnectAuthorizer.OIDCLoginCallback callback, ITokenStore tokenStore) {
   return new OpenIDConnectAuthorizer(callback, tokenStore);
}
```
https://github.com/couchbase/couchbase-lite-java-core/pull/1284/files#diff-b8325d17333d62df13ec5bd23e2b7820R59

maybe we could overrides `AuthenticatorFactory` for Android.